tests/pygeoapi-test-config-apirules.yml

```diff
@@ -0,0 +1,347 @@
+# =================================================================
+#
+# Authors: Tom Kralidis <tomkralidis@gmail.com>
+#          Sander Schaminee <sander.schaminee@geocat.net>
+#
+# Copyright (c) 2019 Tom Kralidis
+# Copyright (c) 2023 Sander Schaminee
+#
+# Permission is hereby granted, free of charge, to any person
+# obtaining a copy of this software and associated documentation
+# files (the "Software"), to deal in the Software without
+# restriction, including without limitation the rights to use,
+# copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following
+# conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+# HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+# WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# =================================================================
+
+server:
+    bind:
+        host: 0.0.0.0
+        port: 5000
+    url: http://localhost:5000/
```

Comment:
What does happen if this url has a root context, i.e. `http://localhost:5000/foo` and the `url_prefix` is set? Can you add a specific test for this?

Reply:
Should be fine, but I will (try to) add a test 👍

Reply:
@francbartoli I'm having trouble testing this. The mock client ignores the `url` setting in the YAML config completely.

In fact, if I set `url` to `http://localhost:5000/api` for example and spin up a Docker container, pygeoapi _does_ start successfully, but it actually runs at http:/localhost:5000 instead. All the links however do have the `/api` suffix, but if I click any of the links, none of them work. I have to remove the suffix to make them work.

The `url` property is only used in api.py and other code to create absolute URLs (which is not very dynamic actually...). Aside: I actually realize that I need to fix all those URL's too if a prefix applies...

**Q:** have **you** ever successfully set a root context using _only_ the `url` property?

In the case of Flask, I could make use of the built-in [`url_for`](https://flask.palletsprojects.com/en/2.2.x/api/?highlight=url_for#flask.url_for) function, which resolves the URL correctly, provided that I set the [`SERVER_NAME`](https://flask.palletsprojects.com/en/2.2.x/config/#SERVER_NAME) and [`APPLICATION_ROOT`](https://flask.palletsprojects.com/en/2.2.x/config/#APPLICATION_ROOT) correctly in the mock client, according to what's defined in pygeoapi's configuration.